sympy/solvers/ode.py

```diff
@@ -395,6 +395,9 @@ def get_numbered_constants(eq, num=1, start=1, prefix='C'):
         raise ValueError("Expected Expr or iterable but got %s" % eq)
 
     atom_set = set().union(*[i.free_symbols for i in eq])
+    func_set = set().union(*[i.atoms(Function) for i in eq])
+    if func_set:
+        atom_set |= {Symbol(str(f.func)) for f in func_set}
```

Comment:
Here's an example of where #15460 would apply: if all functions had a name attribute then this could be `Symbol(f.name)`. But if no function has a name attribute (as is the case in this PR), then what you have done here makes sense and will work even if a non-Undefined functions happened to have a number in the name (like `log2`).

Reply:
@smichr  Yes that makes sense. But looking at the discussion in #15460 I don't think you or @asmeurer  (or in this case, me too) agree to have all the functions return a name attribute. So introducing that change (`Symbol(f.name)`) wouldn't be necessary unless we give all the functions a `name` attribute.

Reply:
@cbm755  I assure you, I'm completely against plagiarism. I did this independently (and actually have been on the receiving end of someone else using my ideas, so I know how it feels and how bad it is) and I therefore appreciate your concern. And yes, whenever I use someone else's commits or work, I first pull them in the PR and then work on top of them and also give them credit for it (you can take a look at a few of my previous PRs)

Reply:
I did not mean to suggest anything inappropriate had happened; please accept my apologies if you felt threatened by my comments.

@asmeurer an equivalent change was done in Diofant (which Github has linked to my original bug report).  We could've simply cherry-picked that change but no one did so.  Instead, we have what I believe to be an independently written solution (based on a different variable name and a clear statement from the author).

I'm now +1 for this change after minor reformating.